diff --git a/libvirt-0.9.12/Makefile.in b/libvirt-0.9.12/Makefile.in
index 4884c59..6aae009 100644
--- a/libvirt-0.9.12/Makefile.in
+++ b/libvirt-0.9.12/Makefile.in
@@ -283,6 +283,17 @@ am__v_at_ = $(am__v_at_$(AM_DEFAULT_VERBOSITY))
 am__v_at_0 = @
 SOURCES =
 DIST_SOURCES =
+
+ifdef GIT_VER
+ G_TAG := --define "_gitVer $(GIT_VER)"
+endif
+ifdef TAG
+ G_TAG += --define "_buildTag $(TAG)"
+endif
+ifdef FILE_LIST
+ G_TAG += --define "_fileList $(FILE_LIST)"
+endif
+
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
 	html-recursive info-recursive install-data-recursive \
 	install-dvi-recursive install-exec-recursive \
@@ -1890,7 +1901,7 @@ distcheck: dist
 	  && am__cwd=`pwd` \
 	  && $(am__cd) $(distdir)/_build \
 	  && ../configure --srcdir=.. --prefix="$$dc_install_base" \
-	    $(DISTCHECK_CONFIGURE_FLAGS) \
+	    --localstatedir=/var $(DISTCHECK_CONFIGURE_FLAGS) \
 	  && $(MAKE) $(AM_MAKEFLAGS) \
 	  && $(MAKE) $(AM_MAKEFLAGS) dvi \
 	  && $(MAKE) $(AM_MAKEFLAGS) check \
@@ -2070,7 +2081,7 @@ $(top_srcdir)/HACKING: $(top_srcdir)/docs/hacking1.xsl $(top_srcdir)/docs/hackin
 	   > $@-t && mv $@-t $@ ; fi );
 
 rpm: clean
-	@(unset CDPATH ; $(MAKE) dist && rpmbuild -ta $(distdir).tar.gz)
+	@(unset CDPATH ; $(MAKE) dist && rpmbuild -ta $(distdir).tar.gz $(G_TAG) )
 
 check-local: all tests
 
diff --git a/libvirt-0.9.12/libvirt.spec.in b/libvirt-0.9.12/libvirt.spec.in
index 2e08abb..6571a3f 100644
--- a/libvirt-0.9.12/libvirt.spec.in
+++ b/libvirt-0.9.12/libvirt.spec.in
@@ -270,10 +270,17 @@
 %define with_rhel5  0
 %endif
 
-Summary: Library providing a simple virtualization API
+%if 0%{?_buildTag:1}
+%define         _relstr      %{_buildTag}
+%else
+%define         _relstr      %(date -u +%y%m%d%H%M)
+%endif
+%{echo: "Building release %{_relstr}\n"}
+
+Summary: Library providing a simple virtualization API  %{?_gitVer}
 Name: libvirt
 Version: @VERSION@
-Release: 1%{?dist}%{?extra_release}
+Release: %{_relstr}
 License: LGPLv2+
 Group: Development/Libraries
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root
diff --git a/libvirt-0.9.12/src/qemu/qemu_command.c b/libvirt-0.9.12/src/qemu/qemu_command.c
index 117542f..813c81a 100644
--- a/libvirt-0.9.12/src/qemu/qemu_command.c
+++ b/libvirt-0.9.12/src/qemu/qemu_command.c
@@ -4929,6 +4929,7 @@ qemuBuildCommandLine(virConnectPtr conn,
 
             if (actualType == VIR_DOMAIN_NET_TYPE_NETWORK ||
                 actualType == VIR_DOMAIN_NET_TYPE_BRIDGE ||
+                actualType == VIR_DOMAIN_NET_TYPE_ETHERNET ||
                 actualType == VIR_DOMAIN_NET_TYPE_DIRECT) {
                 /* Attempt to use vhost-net mode for these types of
                    network device */
diff --git a/libvirt-0.9.12/src/qemu/qemu_hotplug.c b/libvirt-0.9.12/src/qemu/qemu_hotplug.c
index 9b60816..7fb75d3 100644
--- a/libvirt-0.9.12/src/qemu/qemu_hotplug.c
+++ b/libvirt-0.9.12/src/qemu/qemu_hotplug.c
@@ -713,6 +713,9 @@ int qemuDomainAttachNetDevice(virConnectPtr conn,
         iface_connected = true;
         if (qemuOpenVhostNet(vm->def, net, priv->qemuCaps, &vhostfd) < 0)
             goto cleanup;
+    } else if (actualType == VIR_DOMAIN_NET_TYPE_ETHERNET) {
+        if (qemuOpenVhostNet(vm->def, net, priv->qemuCaps, &vhostfd) < 0)
+            goto cleanup;
     }
 
     if (qemuCapsGet(priv->qemuCaps, QEMU_CAPS_NET_NAME) ||
